### PR TITLE
Add merged custom metric

### PR DESCRIPTION
**Pull request by zkdev**
**What this PR does / why we need it**:
Merge `garden_shoots_custom_*` metrics into one `garden_shoots_custom_merged` metric.
`garden_shoots_custom_addon_kubedashboard_total{}` will now be available as `garden_shoots_custom_merged{customizations="addon_kubedashboard_total"}`

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
@wyb1 

**Release note**:

```improvement operator
`garden_shoots_custom_*` suffix is exposed via label `customizations`
```

### CODE REVIEW

File: pkg/template/template.go [hunk 1]
```diff
@@ -94,3 +108,36 @@ func mapType(t Type) prometheus.ValueType {
 		return prometheus.UntypedValue
 	}
 }
+
+func (m *MetricTemplate) sendMergedMetric(vals []float64, labels [][]string, ch chan<- prometheus.Metric) {
+	l := make(map[string]string)
+	l["customizations"] = strings.Replace(m.Name, fmt.Sprintf("%s_", metricShootsCustomPrefix), "", 1)
```
Comment (wyb1) [inline pkg/template/template.go]:
```suggestion
	l["customization"] = strings.Replace(m.Name, fmt.Sprintf("%s_", metricShootsCustomPrefix), "", 1)
```
change the label to be singular

File: pkg/template/template.go [hunk 2]
```diff
@@ -94,3 +108,36 @@ func mapType(t Type) prometheus.ValueType {
...
+	var noLabels = len(labels) == 0
+
+	if noLabels {
+		// TODO  closer look
+		if len(vals) == 0 {
+			return
+		}
+		var desc = prometheus.NewDesc(fmt.Sprintf("%s_merged", metricShootsCustomPrefix), "Collection of all collected customization metrics.", nil, l)
```
Comment (wyb1) [inline pkg/template/template.go]: I think we can drop the `_merged` suffix. Is there any reason that we need it?

Comment (zkdev) [reply pkg/template/template.go]: I added the `_merged` suffix to highlight the difference between the specific version e.g.`garden_shoots_custom_addon_kubedashboard_total` and the generic `garden_shoots_custom`. But sure will remove it.

Comment (zkdev) [reply pkg/template/template.go]: https://github.com/gardener/gardener-metrics-exporter/pull/27/commits/216ab326350b8c200516a548e9e7d61416f87898

**Pull request merged**